src/Metadata/Resource/Factory/AnnotationResourceNameCollectionFactory.php

```diff
@@ -47,8 +47,11 @@ public function create() : ResourceNameCollection
         $includedFiles = [];
 
         if ($this->decorated) {
-            foreach ($this->decorated->create() as $resourceClass) {
-                $classes[$resourceClass] = true;
+            $collection = $this->decorated->create();
+            $context = $collection->getContext();
+
+            foreach ($collection as $resourceClass) {
+                $classes[$resourceClass] = isset($context[$resourceClass]) ? $context[$resourceClass] : true;
```

Comment:
We can use the null coalesce operator introduced in PHP 7:

`$context[$resourceClass] ?? true`